management_api_app/api/dependencies/workspaces.py

```diff
@@ -12,13 +12,17 @@
 from resources import strings
 
 
-async def get_workspace_by_workspace_id_from_path(workspace_id: UUID4 = Path(...), workspaces_repo=Depends(get_repository(WorkspaceRepository))) -> Workspace:
+def get_workspace_by_id(workspace_id: UUID4, workspaces_repo) -> Workspace:
```

Comment:
I guess this method should be async too?

Reply:
This method doesn't need to be async and since there is nothing to await and we call it directly, making it async causes issues -- technically the other one doesn't need to be either but the depends methods tend to have that signature. 

If you want to change the other one to not be async, feel free to play around with that to make sure it works and check in the changes before merging.